yb-voyager/src/srcdb/pg_dump_extract_schema.go

```diff
@@ -136,28 +137,38 @@ func parseSchemaFile(exportDir string) int {
 			sqlType := extractSqlTypeFromComment(delimiterLine)
 			switch sqlType {
 			case "SCHEMA", "TYPE", "DOMAIN", "RULE", "FUNCTION",
-				"AGGREGATE", "PROCEDURE", "VIEW", "TRIGGER", "EXTENSION", "COMMENT":
-				objSqlStmts[sqlType].WriteString(stmts)
+				"AGGREGATE", "PROCEDURE", "VIEW", "TRIGGER", "EXTENSION", "COMMENT", "COLLATION":
+				if utils.ContainsString(pgObjList, sqlType) {
+					objSqlStmts[sqlType].WriteString(stmts)
+				}
 			case "SEQUENCE", "SEQUENCE OWNED BY":
-				objSqlStmts["SEQUENCE"].WriteString(stmts)
+				if utils.ContainsString(pgObjList, "SEQUENCE") {
+					objSqlStmts["SEQUENCE"].WriteString(stmts)
+				}
 			case "INDEX", "INDEX ATTACH":
-				objSqlStmts["INDEX"].WriteString(stmts)
+				if utils.ContainsString(pgObjList, "INDEX") {
+					objSqlStmts["INDEX"].WriteString(stmts)
+				}
 			case "TABLE", "DEFAULT", "CONSTRAINT", "FK CONSTRAINT":
-				objSqlStmts["TABLE"].WriteString(stmts)
+				if utils.ContainsString(pgObjList, "TABLE") {
```

Comment:
Instead of duplicating this check:
- let the code build the `objSqlStmts` as it was doing it earlier
- just process the object types that are mentioned in the `pgObjList` in the subsequent code blocks e.g. change the `if` condition on line 175 as:
   `if !utils.ContainsString(pgObjList, objType) || sqlStmts.Len() == 0 {`